src/contexts/AuthContext.tsx

```diff
@@ -1,42 +1,73 @@
-import { createContext, useContext, useEffect, useState } from 'react'
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from 'react'
+import axios from 'axios'
+import { toast } from 'react-toastify'
 
 type AuthContextType = {
-  user: string | null
+  user: { email: string; name: string } | null
   isLoggedIn: boolean
-  login: (username: string) => void
+  login: (email: string, password: string) => Promise<void>
   logout: () => void
+  isAuthReady: boolean
 }
-const STORAGE_KEY_USER = 'user';
+
+const STORAGE_KEY_USER = 'userInfo'
 
 const AuthContext = createContext<AuthContextType | undefined>(undefined)
 
 export const AuthProvider = ({ children }: { children: React.ReactNode }) => {
-  const [user, setUser] = useState<string | null>(null)
-
+  const [user, setUser] = useState<{ email: string; name: string } | null>(null)
+  const [isAuthReady, setIsAuthReady] = useState(false)
   useEffect(() => {
-    const storedUser = localStorage.getItem(STORAGE_KEY_USER)
-    if (storedUser) {
-      setUser(storedUser)
+    const stored = localStorage.getItem(STORAGE_KEY_USER)
+    if (stored) {
+      const parsed = JSON.parse(stored)
+      setUser({ email: parsed.email, name: parsed.name })
     }
+    setIsAuthReady(true)
   }, [])
 
-  const login = (username: string) => {
-    setUser(username)
-    localStorage.setItem(STORAGE_KEY_USER, username)
+const login = async (email: string, password: string) => {
+  try {
+    const res = await axios.post('http://localhost:3000/api/login', { email, password })
```

Comment:
proxy 설정을 해두었으니 
```suggestion
    const res = await axios.post('/api/login', { email, password })
```

로 해로 될 것 같아요!

Reply:
네! 수정 했습니다!!

Reply:
수정이 안된거 같은데 다시 한번 확인해주시겠어요?